lib/Support/Config/TargetConfig.cpp

```diff
@@ -0,0 +1,338 @@
+/*******************************************************************************
+ * Copyright (c) 2022 - 2024 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+
+#include "cudaq/Support/TargetConfig.h"
+#include "llvm/Support/Allocator.h"
+#include "llvm/Support/Base64.h"
+#include "llvm/Support/CommandLine.h"
+#include "llvm/Support/Debug.h"
+#include "llvm/Support/ErrorOr.h"
+#include "llvm/Support/MemoryBuffer.h"
+#include "llvm/Support/Path.h"
+#include "llvm/Support/PrettyStackTrace.h"
+#include "llvm/Support/Regex.h"
+#include "llvm/Support/SourceMgr.h"
+#include "llvm/Support/TargetSelect.h"
+#include "llvm/Support/ToolOutputFile.h"
+#include "llvm/Support/VirtualFileSystem.h"
+#include "llvm/Support/raw_ostream.h"
+#include <sstream>
+
+#define DEBUG_TYPE "target-config"
+
+using namespace llvm;
+
+namespace {
+static std::unordered_map<std::string, cudaq::config::TargetFeatureFlag>
+    stringToFeatureFlag{{"fp32", cudaq::config::flagsFP32},
+                        {"fp64", cudaq::config::flagsFP64},
+                        {"mgpu", cudaq::config::flagsMgpu},
+                        {"mqpu", cudaq::config::flagsMqpu}};
+}
+
+/// @brief Convert the backend config entry into nvq++ compatible script.
+static std::string processSimBackendConfig(
+    const std::string &targetName,
+    const cudaq::config::BackendEndConfigEntry &configValue) {
+  std::stringstream output;
+  if (configValue.GenTargetBackend.has_value()) {
+    output << "GEN_TARGET_BACKEND="
+           << (configValue.GenTargetBackend.value() ? "true" : "false") << "\n";
+  }
```

Comment:
I'm willing to be overridden here, but there are many instances in this file and other files of unnecessary braces, and I was under the impression that we were following this guidance: https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements
```suggestion
  if (configValue.GenTargetBackend.has_value())
    output << "GEN_TARGET_BACKEND="
           << (configValue.GenTargetBackend.value() ? "true" : "false") << "\n";
```
I'll hold off on littering the review with other comments like this until we make a decision here.

Reply:
Agreed. This code is part of the LLVM part of the CUDA-Q codebase so should follow LLVM coding standards. 

Fixed in https://github.com/NVIDIA/cuda-quantum/pull/1890/commits/6b49f72f2a6e2831a3060526ce3dbc9115e8aeb8